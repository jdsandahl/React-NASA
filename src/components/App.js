import React, { useState } from "react";
import "../styles/App.css";
import Search from "./Search";
import SearchResults from "./SearchResults";

function App() {
  const [searchResults, setSearchResults] = useState([]);

  return (
    <div className="App">
      <Search setSearchResults={setSearchResults} />
      <div>
        {searchResults && searchResults.length > 0 && ( searchResults.map((element) => <SearchResults key={element} image={element} /> ))}
      </div>
    </div>
  );
}

export default App;
