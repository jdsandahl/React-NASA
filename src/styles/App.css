.App {
  display: flex;
  flex-direction: column;
  align-content: center;
  margin: auto;
  max-width: 1530px;
}

.nasa-logo { 
  margin: auto;
  width: 500px;
  max-height: 300px;
}

@media only screen and (max-width: 500px){
  .nasa-logo { 
      width: 300px;
    }
}