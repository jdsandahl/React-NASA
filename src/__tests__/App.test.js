import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App.js";

describe("App", () => {
  
  test("renders correctly against snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('with no search results in state', () =>{  
    it('renders the correct elements', () => {
      const {queryByTestId} = render(<App />);

      expect(queryByTestId("nasa-logo")).toBeTruthy();
      expect(queryByTestId("Search")).toBeTruthy();
      expect(queryByTestId("SearchResults")).toBeFalsy();
    });
  });

});
