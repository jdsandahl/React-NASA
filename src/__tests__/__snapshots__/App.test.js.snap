// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasaLogo"
      class="nasa-logo"
      data-testid="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
      data-testid="Search"
    >
      <form
        class="search-form"
        data-testid="search-form"
      >
        <input
          class="search-input"
          data-testid="search-input"
          placeholder="Search"
          type="text"
        />
        <button
          class="search-btn"
          data-testid="search-btn"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
